cyto_dl/image/transforms/crop_to_seg.py

```diff
@@ -0,0 +1,136 @@
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import torch
+from monai.transforms import RandomizableTransform, Transform
+from skimage.measure import regionprops
+
+
+class CentroidCrop:
+    """Class for cropping patches around passed centroids in an image."""
+
+    def __init__(self, crop_size: Sequence[str], remove_edge: bool = True):
+        self.crop_size = crop_size
+        self.remove_edge = remove_edge
+
+    def centroid_to_slice(self, centroid):
+        # slice across channel dimension
+        slices = [slice(None, None)]
+        for i, c in enumerate(centroid):
+            start = int(c - self.crop_size[i] // 2)
+            end = int(c + self.crop_size[i] // 2)
+            slices.append(slice(start, end))
+        return tuple(slices)
+
+    def _filter_edge(self, centroids, shape, labels=None):
+        assert len(shape) == len(
+            self.crop_size
+        ), "Image shape and crop_size must have the same dimensionality"
+        valid_indices = [
+            idx
+            for idx, c in enumerate(centroids)
+            # check distance to edges
+            if all(
+                [
+                    c[i] >= self.crop_size[i] // 2 and c[i] <= (shape[i] - self.crop_size[i] // 2)
+                    for i in range(len(c))
+                ]
+            )
+        ]
+        valid_centroids = np.array(centroids)[valid_indices]
+        if labels is None:
+            return valid_centroids, None
+        return valid_centroids, np.array(labels)[valid_indices]
+
+    def __call__(
+        self,
+        data: Union[np.ndarray, torch.Tensor],
+        centroids: Sequence[Sequence[int]],
+        labels: Optional[Sequence[int]] = None,
+        name="data",
+    ):
+        # don't include channel dimension in edge validation
+        centroids, labels = self._filter_edge(centroids, data.shape[1:], labels)
+        if len(centroids) == 0:
+            raise ValueError("No valid centroids found")
+        crops = [{name: data[self.centroid_to_slice(c)], "centroid": c} for c in centroids]
+        if labels is not None:
+            for crop, label in zip(crops, labels):
+                crop[name] = label
+        return crops
+
+
+class CentroidCropd(CentroidCrop, Transform):
```

Comment:
Maybe `CentroidCropDict` is a better name?

Reply:
Adding the d at the end instead of dict is just to be consistent with monai naming conventions